### DISCUSSION TITLE
LBB on a Pi5 with 2TB internal storage.

### DISCUSSION THREAD

**Discussion by PioB-1999** [outdoorbits/little-backup-box#278] [Show and tell]
Just installed the LBB on a Pi5 with internal 2TB M.2 SSD.
It works like charm with a powerbank. I rotated the screen so it fits better the 4.3" touch screen.
Copying 520 Files (14.6GB) took only 1:12min.
![Front](https://github.com/outdoorbits/little-backup-box/assets/164955427/039b3069-b6ac-447d-a571-e315e5869c95)
I am really excited.
Thanks a lot!
Pio

**Comment by outdoorbits** [comment]
This looks great! Could you please provide more information? 
I have a lot of questions:
Did you design the case? Is it available anywhere?
Witch display are you using? 
What did you do change display direction? (Is this done by gui or by cmd line? - If I knew how to rotate from cmd line I could add this to the config options.)
Did you change anything to use the M.2 SSD? I didn't find the time yet to get and test this. Do you boot from ssd, so you backup just to internal?
And ... cloud I use your image into the gallery?
Thank you!

**Comment by ruebyi** [comment]
It’s almost perfect - If we were able to scroll easier via touch… , but I think that’s a OS Issue

**Comment by PioB-1999** [comment]
Hi,
Where to begin? First of all this are the used components:
1. Raspberry Pi 5 with cooler and latest "Bookworm" OS
2. Waveshare 4.3" Touchdisplay.
3. Geekworm X1002 M.2 NVMe SSD PCIe Board
4. Crucial P3 Plus 2TB M.2 PCIe Gen4 NVMe SSD
5. Raspberry Pi Display Cable Standard - Mini 200mm
6. 10mm M2,5 spacers
7. Self designed 3D printed case

I had some problems at the beginning but with help from friends and vendor support I could save them all.

Here are the settings I used:
1. Switch PI back to X11 and allow VNC. This helped a lot during debugging.
2. Set PCI to Gen 3. This speed up the SSD from 400MB/s to 800MB/s.
3. Transfer the boot SD-Card to internal SSD, so the PI boots direct from SSD.
4. Display rotation: add `@xrandr --output DCI-2 --rotate left` to /etc/xdg/lxsession/LXDE-pi/autostart
5. Touch rotation: add `Option "CalibrationMatrix" "0 -1 1 1 0 0 0 0 1"` to /usr/share/X11/xorg.conf.d/40-libinput.conf into touchscreen section.
6. Installed also the brightness tool from display vendor to control the brightness.
7. Connected my Logi MX Keys and MX Master via BT.

With my first case design there was a problem with getting the scroll bar touched. I changed my design and now it is much easier.
But I must admit using the BT mouse is easier to scroll.

Here are the newest pictures:

![Front](https://github.com/outdoorbits/little-backup-box/assets/164955427/6d0ac5ec-826c-4181-b624-6fbf386731ab)
![Back](https://github.com/outdoorbits/little-backup-box/assets/164955427/fa706f1d-2efc-4b02-b2f3-e42312256bc3)
![Inward](https://github.com/outdoorbits/little-backup-box/assets/164955427/744772d3-ceea-44bd-a67c-7a0942d5c3aa)

You can put them into the gallery.

What I was missing during my testing was the possibility to stop and start the LBB from the PI desktop.
Finally I found a way to do it via VNC.
But I´m a windows guy and this was my first attempt to use a Raspberry OS in that way. I must learn a lot.

The reason building this LBB was my upcoming trip to Tanzania.

Greetings
Pio

**Reply by phil-Xbones** [reply]
Hello  
I'm  new on the community and wants to  thanks all the greats developpers for this famous LBB 
very great Job  . I was using old style LBB with small oled screen and decided to build this very great  version , unfortunately I'm not a raspberry guy ( more arduino ;-)  and I got issue to rotate the screen on my Pi 4 and the same screen as yours , and  finally manage  to did it with the command @xrandr --output DSI-1 --rotate left to /etc/xdg/lxsession/LXDE-pi/autostart . but since I install a PI 5 and I reprog it , no way to have the left rotation  even with your command @xrandr --output DCI-2 --rotate left to /etc/xdg/lxsession/LXDE-pi/autostart   this is my first pb  but I will manage . today  my 2 big concern are : 
-1) no way to finalize the Thumbnails  generation , nor after the backup, nor manually with the  "view database"  menu  generate thumbnails , it stops generating the thumbnails with not reason in the log files , I tried several different storage , even my sandisk extreme 2T portable SSD which was working great with my first LBB 
- 2) I also install a nvme pci drive same than yours but 1 T , are you able to make the backup on the internal NVMe drive ? I can see it  with ssh lspci , but I  can't select the nvme drive as target .

thanks for any help 
phil

**Comment by outdoorbits** [comment]
The touchscreen rotation seems to be device specific? Display rotating didn't work on a short try here. Maybe this should be done hardware specific manually.
If you (or anybody else) finds out how to setup that hardware independent by command line, I would try to integrate that into settings.

**Reply by outdoorbits** [reply]
For your journey, please keep in mind a power cut could damage your operating system. That means, it's safer to boot from SD and carry at least one more installed lbb system with you on SD.
Therefore I have to add another device option to address the SSD as target or source. This time I don't have a M.2 SSD so I can't test this myself. I would need your help. This means just to execute a few commands on command line and test the function. Would you help?

**Reply by PioB-1999** [reply]
Are you using X11 or Wayland? XRANDR Rotation works only with X11.
I will use the LBB only with a powerbank, so a power cut is improbable.
But never mind I will take a LBB SD with me.
Yes, I can help you testing the SSD. Send me the commands or call me.

**Comment by PioB-1999** [comment]
Here the Link to my case: https://www.printables.com/model/824210-case-for-raspberry-pi5-with-43-display-and-2tb-nvm

**Reply by Owz81** [reply]
Hi,
I try to rebuild your project. It seems that waveshare changed the mounting points so the ports on USB/Ethernet side are way to deap in the case and the posts on Power/HDMI side are not alline with the ports any more. 
![20240614_190658](https://github.com/outdoorbits/little-backup-box/assets/109694275/958e1c12-d7b8-48f1-85cd-f8ff5dc18fd3)
![20240614_192056](https://github.com/outdoorbits/little-backup-box/assets/109694275/f18768ec-1985-4bde-b1b2-2d5aa1ab6c72)
![20240614_190641](https://github.com/outdoorbits/little-backup-box/assets/109694275/32994409-43c7-4292-acb2-c55800ccbfb1)

**Reply by PioB-1999** [reply]
How did you placed the RPi on the screen? 
In my configuration it is rotated by 180°. It is not like marked out on the screen  background.
What are the dimensions of your print. My is 117.7mm wide and 78.4mm deep.

**Reply by Owz81** [reply]
I will try the other way this evening. Has to help my Sister today, 
![20240615_100759](https://github.com/outdoorbits/little-backup-box/assets/109694275/4e4de73f-4770-4bfa-a4d2-a239c3d31086)

**Reply by Owz81** [reply]
My print is a little bit smaller: 117,5mm x 78,3mm

**Reply by Owz81** [reply]
If i turn the Pi it fites much better. But I see no way to get the Cabel to the Pi without destroiing it of cover one of the HMDI Ports...
![20240615_185720](https://github.com/outdoorbits/little-backup-box/assets/109694275/54fdb97e-ab06-413b-809d-800890081c75)

**Reply by PioB-1999** [reply]
It seems you need a longer cable.

**Reply by Owz81** [reply]
I have a Camera Kabel 200mm laying around. Fits great, but has a different pinnig to the Waveshare cable...
Do you know were I can find longer cable with the Waveshare pinnig?

**Reply by Owz81** [reply]
OK! I find one on Amazon. Next try on Tuesday!

**Reply by Owz81** [reply]
With the longer cable, it seems to work. But no touch function any more on the display...

**Comment by outdoorbits** [comment]
I linked it back from the gallery.

**Comment by outdoorbits** [comment]
I used Wayland. But I really think automation of screen configuration 
would exceed any resources.

Meanwhile I included NVMe into lbb. But I can't test it yet - but parts 
are ordered now. To test it you can just update. 

But (if there is no error) you will find no difference if you booted 
from the NVMe device. If you boot from SD you'll get extra sources and 
targets. I'm excited if it works... ;-)

**Comment by tutle** [comment]
Hello, I'm very interested in this configuration to change my "old" faithfull [nsb25 nexto](https://zxrecords.com/kit/nexto-di-nsb-25-backup-device/) (no more support, discontinued and proprietary system...), I would like to know if there is a sync option in LBB in case of reinsert a previous card or update an external backup ssd.
The nexto is so easy to use in travel mode (automatic folder, sync...), I would like as friendly as possible (but with better transfer rates)

**Comment by outdoorbits** [comment]
Lbb does a sync, pre existing files at target are not transferred again. 
Backups are done into source device specific folders at target.

**Comment by PioB-1999** [comment]
I have finally found a power bank that can supply 5A at 5V. This is important, otherwise the RPi 5 will run at a throttle.
It is the Shargeek 100 https://sharge.de/products/storm2
![IMG_0906](https://github.com/outdoorbits/little-backup-box/assets/164955427/b4f41e73-5c78-4b46-8c9b-1411ed7f0109)
![IMG_0907](https://github.com/outdoorbits/little-backup-box/assets/164955427/662c12a5-6912-4c28-8bf1-8a607dd95717)
![IMG_0908](https://github.com/outdoorbits/little-backup-box/assets/164955427/5aab866a-34da-49d8-8230-6fd7f9e0242b)

I also tested the Waveshare Uninterruptible Power Supply Module https://www.waveshare.com/ups-module-3s.htm
But it has no display and supports only RPi. 
![IMG_0909](https://github.com/outdoorbits/little-backup-box/assets/164955427/3fe9a884-ea59-4059-a174-363ad215138a)

**Comment by outdoorbits** [comment]
I can't help you about the screen rotation but there are some guys 
having more experience about that.

1) Please ssh to your pi and test this:
`sudo python /var/www/little-backup-box/backup.py --SourceName thumbnails --TargetName usb --power-off False`
Is there an error message?

2) What do you get for
`sudo python3 /var/www/little-backup-box/lib_storage.py --Action get_available_partitions --skipMounted False --ignore-fs True`

I would ask you, to open a new discussion on every single problem, this 
makes it easier to follow the discussion for anybody later.

**Reply by phil-Xbones** [reply]
thnaks so much for your fast feed back 

for point 1 , here is the answer :
sudo python /var/www/little-backup-box/backup.py --SourceNa                                                                                                                                                             me thumbnails --TargetName usb --power-off False
Traceback (most recent call last):
  File "/var/www/little-backup-box/backup.py", line 1288, in <module>
    backupObj.run()
  File "/var/www/little-backup-box/backup.py", line 211, in run
    self.generateThumbnails()
  File "/var/www/little-backup-box/backup.py", line 980, in generateThumbnails
    db.dbInsertImage(SourceFilePathName)
  File "/var/www/little-backup-box/lib_view.py", line 190, in dbInsertImage
    self.__cur.execute(f"alter table EXIF_DATA add column {EXIF_Field} text;")
sqlite3.OperationalError: near ".": syntax error

for second point I got his 👍  sudo python3 /var/www/little-backup-box/lib_storage.py --Action get_available_partitions --skipMounted False --ignore-fs True
/dev/sda: /dev/sda
/dev/sda1: --uuid 10CE6071CE605150
/dev/nvme0n1: /dev/nvme0n1



i will open  the 2 new discussion you are right 

phil

**Reply by phil-Xbones** [reply]
for display rotation and touch pad rotation here is what I did for 4.3 screen DSI LCD waveshare 800x480

rotation  touch pad :

1. Install libinput:

sudo apt-get install xserver-xorg-input-libinput

If the system you installed is an Ubuntu system, or Jetson Nano, the installation command is:
sudo apt install xserver-xorg-input-synaptics

2. Create the xorg.conf.d directory under /etc/X11/ (if this directory already exists, go directly to step 3):

sudo mkdir /etc/X11/xorg.conf.d
3. Copy "40-libinput-conf" file to the directory just created:

sudo cp /usr/share/X11/xorg.conf.d/40-libinput.conf /etc/X11/xorg.conf.d/
4. Edit:

sudo nano /etc/X11/xorg.conf.d/40-libinput.conf 
Find the touchscreen part, add the following sentence, and then save.

#Find the touchscreen part, add the corresponding rotation Angle command, and then save:  
#90°Right touch rotation: 
#Option "CalibrationMatrix" "0 1 0 -1 0 1 0 0 1"
#180° Inverted touch rotation:
#Option "CalibrationMatrix" "-1 0 1 0 -1 1 0 0 1"
#270° Left rotation:
Option "CalibrationMatrix" "0 -1 1 1 0 0 0 0 1"       



rotation Display ( vertical)

sudo nano  /etc/xdg/lxsession/LXDE-pi/autostart
add :   @xrandr --output DSI-1 --rotate left  works with  PI 4 et PI 5 cable plugged on display 0


phil

**Comment by davewoe** [comment]
I have no idea why - but on my new RPi 5 I can not seem to get the LBB software to install.  I have copied and pasted the curl -sSL line from the instructions... and although I get no error... I'm also not getting install or update information.  Any idea what's going on?

**Reply by outdoorbits** [reply]
There should be an install-error.log file in /home/pi. Could you please post it here?

**Reply by davewoe** [reply]
there isn't... I don't think it installed **AT ALL**

**Reply by davewoe** [reply]
![ssh](https://github.com/user-attachments/assets/f54272ba-d1a1-450a-a758-dc312f39225b)
I've shown my install - but it's not run anything afterwards

**Reply by davewoe** [reply]
![20250419_151805](https://github.com/user-attachments/assets/4dfc303a-1c54-430d-b631-eca4b9b53486)
what it shows under the install-error.log file

**Comment by outdoorbits** [comment]
But you didn't copy the full line?
` branch='main'; curl -sSL https://raw.githubusercontent.com/outdoorbits/little-backup-box/${branch}/install-little-backup-box.sh | bash -s -- ${branch} 2> install-error.log`
You need to set branch.

**Comment by tutle** [comment]
Hello, I really need to build a new unit , my old (faithfull) nsb25 is definively too slow, with amount of ram do you recomend with a Pi 5 ? 4 or 8 ? have a good week end

**Comment by outdoorbits** [comment]
A pi 5 is a fantastic idea for lbb, especially if you want to use the image viewer. In my eyes the RAM size is not really relevant in this case - 4 GB are sufficient.

**Reply by tutle** [reply]
Thanks for this answer, I wanted to know, before starting, if it's possible to make simultaneous backup on internal nvme and a connected hard drive .
I'll send pictures of my buil unit once done.

**Comment by outdoorbits** [comment]
You want to backup two sources to two different targets at the same time? I'm sorry this does not work. But if you have one target, you can backup all connected sources automatically without intervention. 
I'm excited to see your build!

**Comment by tutle** [comment]
No, I'm wanting to make directly a double copy of one source (for safety and futur dispatch of rushes with the director-editor)

**Comment by outdoorbits** [comment]
I'm sorry, this would not work in one step. What I do on traveling is to backup my primary target from time to time to a secondary target. In this case, you can just plug your target first and the source second, like every backup...

**Comment by tutle** [comment]
Ok, a litle sad, I had this so usefull option on my nsb25, the possibility to multi backup in once and also to sync an external drive. Any possibility to implement this ?
Hurry to build one LBB anyway, my nsb is so slow 200gb, 35 minutes (but on 2 drives same time)
<img width="1024" height="768" alt="nsb25 backup" src="https://github.com/user-attachments/assets/84f2a567-4564-4cd1-bda1-0ea94af38b44" />

**Comment by outdoorbits** [comment]
Little Backup Box is designed to work as robustly as possible. Depending on the backup mode, the target device is either the first connected USB device, the NVMe, or a cloud target. All other connected devices automatically become sources. This works reliably in every conceivable variant: without a display, with small LCD displays and with touchscreens.
Simultaneous backup to multiple targets would break this simple concept. You would either have to interact every time and define the targets, or you would have to place corresponding config files on the target drives. That would increase complexity and therefore also the risk of errors.
On top of that, even the Raspberry Pi 5 is fairly limited on USB, and transfer speed would likely suffer.
If you want a second identical copy of your source, you can achieve this with LBB in two steps: first back up the source to target 1, then plug in target 2 and copy target 1 to target 2 (storage mode). That covers your use case (safety copy + handing rushes over to the director-editor) without breaking the simple one-target concept.

**Comment by tutle** [comment]
Hello, I'm not so far from ending my unit,...
Just finished hardware, the print box is on the drawing way, (I had to draw the 1001 board and adapt the 1200 4 cells board ...)
I have a couple of question, as I'm installing the software today :

- _Switch PI back to X11 and allow VNC_. **how do you do this ?**

-_Set PCI to Gen 3. This speed up the SSD from 400MB/s to 800MB/s._
add in /boot/firmware/config.txt
dtparam=pciex1
dtparam=pciex1_gen=3
that's it ?

- What are the average best transfers rates possible from USB to USB ?
- Is there a possible way to connect the LBB to a computer as an hard drive ?

- _Installed also the brightness tool from display vendor to control the brightness_._
I have this [4.3 touch display](https://www.waveshare.com/4.3inch-dsi-lcd.htm)

- I would like to install power [enhancement for 1200 board](https://wiki.geekworm.com/index.php?title=X1202&mobileaction=toggle_view_desktop#Script_Reference), do I do it before installing LBB ?

<img width="1920" height="1080" alt="BRD1" src="https://github.com/user-attachments/assets/ec404123-232b-4f43-b4da-08473477adab" />
<img width="806" height="611" alt="{51406263-D171-43D2-BE85-B2502E64ACCA}" src="https://github.com/user-attachments/assets/aab32ac7-79ad-499f-b0a1-5ddd67abe274" />
<img width="988" height="708" alt="{1A76CA19-8E58-44AE-9DAB-E29C12319736}" src="https://github.com/user-attachments/assets/2d19e6d7-6621-47c2-88b9-bf5d863349ef" />

**Reply by tutle** [reply]
Tried a few install, with Trixie 64, I was able to have webGui control but no local screen control.
With bookworm, no web and no screen.
I see on [this video](https://youtu.be/qAEebY8UlRc?si=GLnmPK32lQ6OWa9s&t=321), Pi4 Os seems lighter even on Pi5 but PCI3 stills possible ?

**Reply by outdoorbits** [reply]
Looks great. I'm in holiday this week, next week I'll try to help. At first I would recommend to use Trixie, going back to bookworm has huge disadvantages and will not work with actual lbb.

**Reply by tutle** [reply]
I was finally able to start the lbb and made a transfer test from cf express B usb to ssd usb about 300mo/s
My nvme is not recognised when I` sudo lspci` :( to make internal test
Will reinstall all with trixie.

**Reply by tutle** [reply]
Hello, some news, 
nvme problem was an inversion of the pci cable (didn't notice there was a direction as it finish same each side).
Trixie installed, can only connect to LBB through webgui from another pc, I don't have LBB interface on the Pi.
Limitation of transfer, enable pci 3 on nvme is not very usefull as the PI5 has only usb 3.0 ports.
So maybe an option to benefit of pci gen3 rates should be not to have internal storage but an [USB 3.2 pci HAT](https://monraspberry.com/produit/carte-dextension-pcie-vers-m-2-4g-5g-et-usb-3-2-pour-raspberry-pi-5/).
Question : would an usb 3.2 HAT be supported by LBB ?
I would really like to have good rates as it's for video shooting.

**Comment by outdoorbits** [comment]
Lbb does not care about the USB version as long as the port is provided by the OS. I would expect it to work out of the box, maybe you need a driver software.

**Reply by tutle** [reply]
Ok, I will maybe have a try of transfer with a board [like this](https://www.waveshare.com/wiki/PCIe_TO_Gigabit_ETH_USB3.2_HAT+). but maybe if 2 usb 3.2 connected, the benefit won't be evident.
Strange, I can't have lbb interface direct on my unit..searching.
Enjoy your holidays. (mute notifications ;-)

**Reply by tutle** [reply]
An ultimate question, do you think it could be possible to have a preview of videos in gallery ?

**Comment by outdoorbits** [comment]
You can view videos in single view. Do you need this in grid view too?

**Reply by tutle** [reply]
I can't read vidéo in single view and rating/viewing in grid should be great.

**Comment by outdoorbits** [comment]
Video in the grid might be an overload for transfer. But why does single view not play your videos? What's your video format and your file extension?

**Reply by outdoorbits** [reply]
Did you get any further?

**Reply by tutle** [reply]
Hello, sorry, I was busy these times.
My video are GH6 log mov (need a codec witch vlc has for example).
I'm thinking to maybe rebuild a new unit with a [Lattepanda Iota](https://www.lattepanda.com/lattepanda-iota) witch is much more powerfull and interesting than PI5 and has real pci 3 and 3 usb 3.2 gen 2.
Transfer rate should be really better. 
Do you think LBB can be use on it ?

**Comment by outdoorbits** [comment]
If it's a Debian OS and you accept to lose all gpio functions like display and buttons, it should be possible to adapt it.
